nncf/experimental/torch/fx/transformations.py

```diff
@@ -0,0 +1,398 @@
+    by outputs of corresponded module call.
+
+    :param module_to_insert: Given torch.nn.Module to insert.
+    :param target_points: Target points to insert the target module.
+    :returns: Transformation which inserts given module to a target model and calls given module
+        after each target points. For each target node all original ouputs
+        are being replaced by outputs of corresponded module call.
+    """
+
+    def module_insertion_transformation(model: torch.fx.GraphModule):
+        module_attr_name = _set_module_to_the_graph_module(model, module_to_insert, target_points)
+        graph = model.graph
+        for target_point in target_points:
+            target_node = _get_target_node(graph, target_point)
+            new_node = _insert_call_module(graph, target_node, module_attr_name)
+            for user in list(target_node.users):
+                if user is new_node:
+                    continue
+                user.replace_input_with(target_node, new_node)
+
+    return module_insertion_transformation
+
+
+def leaf_module_insertion_transformation_builder(
+    module_to_insert: torch.nn.Module, target_points: List[PTTargetPoint]
+) -> TransformationFNType:
+    """
+    Returns transformation which inserts given module to a target model
+    and calls given module after each target points.
+
+    :param module_to_insert: Given torch.nn.Module to insert.
+    :param target_points: Target points to insert the target module.
+    :returns: Transformation which which inserts given module to a target model
+        and calls given module after each target points.
+    """
+
+    def leaf_module_insertion_transformation(model: torch.fx.GraphModule):
+        module_attr_name = _set_module_to_the_graph_module(model, module_to_insert, target_points)
+        # Insert call_module nodes to the model
+        graph = model.graph
+        for target_point in target_points:
+            target_node = _get_target_node(graph, target_point)
+            _insert_call_module(graph, target_node, module_attr_name)
+
+    return leaf_module_insertion_transformation
+
+
+def bias_update_transformation_builder(node: NNCFNode, value: torch.Tensor) -> TransformationFNType:
+    """
+    Return transformation which updates constant of the given bias node to the given value.
+
+    :param node: Bias node which requires bias constant update.
+    :param value: New value to use as the bias constant.
+    :return: Transformation which updates constant of the given bias node to the given value.
+    """
+
+    def bias_update_transformation(model: torch.fx.GraphModule):
+        graph = model.graph
+        target_node_name = node.node_name
+        graph_node = FXModelTransformer.get_graph_node_by_name(graph, target_node_name)
+        bias_node = next(iter(graph_node.users))
+        with graph.inserting_before(bias_node):
+            new_constant = create_getattr_from_value(model, graph, target_node_name + "_shifted_bias", value)
```

Comment:
would it be suitable to use `setattr` instead to change the bias value of the bias node?

Reply:
Bias value is kept as a torch module buffer, it is not possible to use `setattr` to set value of a torch buffer. To do it correctly one need to register new buffer, create a `getattr` node, replace all usages of the old buffer node with new one and call `graph.eliminate_dead_code()` to release unused old buffer

Reply:
Oh, alright! I thought it would be more convenient by directly changing instead of updating the list of arg nodes but did not factor this in.